libraries/aws_iam_policies.rb

```diff
@@ -26,7 +26,14 @@ class AwsIamPolicies < AwsResourceBase
 
   def initialize(opts = {})
     super(opts)
-    validate_parameters
+    validate_parameters(allow: %i(only_attached scope))
+    @only_attached = opts[:only_attached]
+    @scope         = opts[:scope]
+    @path_prefix   = opts[:path_prefix]
+    @parameters = {}
+    @parameters[:only_attached] = @only_attached if @only_attached
+    @parameters[:scope]         = @scope         if @scope
+    @parameters[:path_prefix]   = @path_prefix   if @path_prefix
     @table = fetch_data
```

Comment:
Does parameters need to be global here? It could be passed into the fetch_data method.
@table = fetch_data(parameters)